tools/testinfra/runner/src/main.rs

```diff
@@ -226,6 +280,92 @@ fn report<P: AsRef<Path>>(tests: Vec<TestResult>, path: P) -> Result<()> {
     return Ok(());
 }
 
-fn xml_escape_text(unescaped: String) -> String {
+fn xml_escape_text(unescaped: &String) -> String {
     return unescaped.replace("<", "&lt;").replace("&", "&amp;");
 }
+
+fn query_disabled(db: &mut Option<Client>) -> HashSet<(String, Option<String>)> {
+    // we have unit test name as NOT NULL in the DB, so empty SQL strings are
+    // converted to Option types in Rust. the inverse should be done when inserting
+    match db {
+        None => HashSet::new(),
+        Some(db) => db
+            .query("SELECT target, test FROM tests WHERE disabled = true", &[])
+            .unwrap()
+            .into_iter()
+            .map(|row| {
+                let target = row.get("target");
+                let test = row.get("test");
+                let test = if test == "" { None } else { Some(test) };
+                return (target, test);
+            })
+            .collect(),
+    }
+}
+
+fn query_commit(db: &mut Option<Client>, revision: String, tests: &Vec<TestResult>) -> Result<()> {
```

Comment:
This name is a bit misleading, I'd rename it to something more like `record_test_results`. `query_*` usually implies a readonly query that is just fetching some information